tests/components/tado/test_config_flow.py

```diff
@@ -78,6 +78,80 @@ async def test_form_invalid_auth(hass: HomeAssistant) -> None:
     assert result2["errors"] == {"base": "invalid_auth"}
 
 
+async def test_form_key_error(hass: HomeAssistant) -> None:
```

Comment:
1. Are these tests covering something new in the dependency bump?
2. Please make sure all config flow tests are completing successful to show that they can recover from a fault.
3. Since only the side effect changes, maybe checkout `@pytest.mark.parametrize` as you can declare the test only once and only swap out the side effect and the returned base error

Reply:
Thanks for the review, Joost! To answer your review I also included a new commit to improve the tests with your suggestion to paramterize it.

1. The coverage is more a gesture to improve the overall code quality of the config flow. It was 83% and I upgraded it to 96%. There is still some code left, which I believe it not touched. Maybe the code owners can take a look into the function `_username_already_configured`.
2. Isn't this already done in the original `test_form`? Maybe I don't understand, so feel free to elaborate your point. :)
3. Thanks! That reduced some code and it has been patched.

Reply:
If possible could you maybe add the tests in a separate PR? That way this PR gets smaller and easier to review.

about 2: We would like the tests testing config flow failures to show that they are able to be completed normally, so they don't cause some kind of state issue (for example if causing an error sets another value which makes it impossible to complete the flow normally)

Reply:
Sure, the tests are restored to their original ones. Hence, this reduces the code coverage back to 83% on the config flow. I don't creating a dedicated PR to rewrite the tests to fits the standards.